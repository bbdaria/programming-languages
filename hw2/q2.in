print "===TEST START===\n";
curry op*;
val test_1 = if 12 = it 3 4 then "Passed" else "Failed";
uncurry it;
val test_2 = if 12 = it (3,4) then "Passed" else "Failed";
