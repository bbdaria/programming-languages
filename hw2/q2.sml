fun curry f a b = f(a,b);
fun uncurry f(a,b) = f a b;