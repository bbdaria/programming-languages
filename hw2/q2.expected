val it = () : unit
val it = fn : int -> int -> int
val test_1 = "Passed" : string
val it = fn : int * int -> int
val test_2 = "Passed" : string
- 
